brave/brave-core
# Update omnibox search conversion promotion banner UI

fix brave/brave-browser#31746

We'll use four types of banner and select one type via griffin.
When user choose "maybe later" button, promotion entry is not shown
till 3d passed and then show it again.

Light theme (type A):
<img width="1119" alt="image" src="https://github.com/brave/brave-core/assets/6786187/0db3014b-e4da-4e0b-b0af-626840731f95">

Dark theme (type B):
<img width="1106" alt="image" src="https://github.com/brave/brave-core/assets/6786187/abbddafa-a567-43f9-bf31-c7504085db50">

type C:
<img width="1119" alt="image" src="https://github.com/brave/brave-core/assets/6786187/ab2b913e-983e-4710-914f-f0a76d20095c">

type D:
<img width="1105" alt="image" src="https://github.com/brave/brave-core/assets/6786187/9b9d33b1-e4cc-415d-834f-54e12971521b">



Resolves 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`OmniboxPromotionTest.*`
`BraveSearchConversionTest.*`

1. Clean profile and re-launch official brave with `--variations-server-url=https://variations.bravesoftware.com/seed` till brave://version has `BraveSearchPromotionBannerStudy` is shown
2. Set google for default search engine
3. Type any keywords and check promotion banner (see above) is shown as a last omnibox entry
4. Click maybe later and check promotion entry is not shown till 3days passed.
5. Terminate browser
6. For quick test, delete last digit of `maybe_later_clicked_time` from `Preferences` file in user dir to simulate enough times passed
7. Launch with `--variations-server-url=https://variations.bravesoftware.com/seed`
8. Type any keywords and check promotion banner(see above) is shown as a last omnibox entry
